### ISSUE TITLE
configure fails because conftest.c doesn't exist

### ISSUE THREAD

**Issue by amoh-godwin** [PortAudio/portaudio#590]
**A piece of the error log reads:**
```
configure:3153: checking whether the C compiler works
configure:3175: C:\MinGW\bin\gcc.exe    conftest.c  >&5
gcc.exe: error: CreateProcess: No such file or directory
configure:3179: $? = 1
configure:3217: result: no
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME ""
| #define PACKAGE_TARNAME ""
| #define PACKAGE_VERSION ""
| #define PACKAGE_STRING ""
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
| /* end confdefs.h.  */
| 
| int
| main ()
| {
| 
|   ;
|   return 0;
| }
configure:3222: error: in `/h/CS/build/portaudio/portaudio':
configure:3224: error: C compiler cannot create executables
See `config.log' for more details
```
I checked to see if conftest.c exists, it didn't. So I decided to create one with the code that is above, copied and pasted it and removed the '|'. When I run `bash-3.1$ ./configure` it deleted the conftest.c and still printed the same error.
Also notice the multiple spaces between C:\MinGW\bin\gcc.exe and conftest.c  `C:\MinGW\bin\gcc.exe    conftest.c`

**Desktop:**
 - OS: Windows
 - OS Version: 10
 - PortAudio version: pa_stable_v190700_20210406

**Comment by RossBencina** [comment]
The configure script should generate the test .c file from the script. It is not missing.

**Comment by RossBencina** [comment]
> C:\MinGW\bin\gcc.exe conftest.c

Ok, but is gcc.exe in your PATH? you need to be able to run gcc.exe without any path qualifer. That's what I think the error is saying: it can't find your compiler. Check config.log for details.

**Comment by amoh-godwin** [comment]
@RossBencina I did I've added it. You can run gcc at the prompt at both the native cmd and also at the bash prompt.
But I get the same error, the same error log.

**Comment by RossBencina** [comment]
I think this is the problem line:

> gcc.exe: error: CreateProcess: No such file or directory

Possibly this fix? https://stackoverflow.com/a/6085411/2013747

**Comment by amoh-godwin** [comment]
Same error.
gcc is the one reporting the error. Check the log, third line from the one I posted. The conftest.c is just not there

**Comment by npostavs** [comment]
If `conftest.c` was missing, you would get an error like this:
```
$ gcc conftest.c
gcc.exe: error: conftest.c: No such file or directory
gcc.exe: fatal error: no input files
```

The `CreateProcess` failure suggests some exe is missing.

**Comment by amoh-godwin** [comment]
> configure:3175: C:\MinGW\bin\gcc.exe    conftest.c  >&5
> gcc.exe: error: CreateProcess: No such file or directory

@npostavs well that is the exact same error.

**Comment by npostavs** [comment]
@amoh-godwin no, one mentions `conftest.c` by name, where the other just has `CreateProcess`.

**Comment by amoh-godwin** [comment]
@npostavs oh okay

**Comment by RossBencina** [comment]
> The CreateProcess failure suggests some exe is missing.

Or a dll that it links to.

**Comment by philburk** [comment]
Can you clarify what "fails" means.
Perhaps MinGW is not installed correctly?

Does ./configure correctly determine that MinGW is broken?
Or does ./configure crash and not create a Makefile?

**Comment by amoh-godwin** [comment]
@philburk I have updated my first comment to include the entire log.

**Comment by RossBencina** [comment]
We'd be iterested to hear whether anyone else is having this issue.

**Comment by RossBencina** [comment]
I will retest this once #994 has landed

**Comment by RossBencina** [comment]
I think I'm seeing this with CI in #1047 now as well, need to cross-check.

#1020 might fix this. need to recheck once merged. [NOPE didn't fix it]

**Comment by RossBencina** [comment]
@amoh-godwin Here is something to try. I have been working on getting native Windows MinGW builds working for CI. I discovered that MSYS2 recommends re-running `autoreconf` (from an MSYS2 shell) as follows:[1]

```
$ pacman -S "${MINGW_PACKAGE_PREFIX}-autotools"
$ pacman -S "autoconf-archive"
$ autoreconf -fvi
```

Then, as usual:

```
$ ./configure
$ make
```

Where `${MINGW_PACKAGE_PREFIX}` might be, for example, `mingw-w64`.

You probably already have the `${...}-autotools` package installed (the `gcc` package requires it), but you may not have the `autoconf-archive` package installed, but it is needed.

Using these commands PortAudio is now successfully configuring and building in GitHub CI on Windows with mingw:

https://github.com/PortAudio/portaudio/actions/runs/16821083986/job/47647942802?pr=1055

[1] https://www.msys2.org/docs/autotools/

**Comment by amoh-godwin** [comment]
@RossBencina I appreciate all the work you are putting into this. I am no longer using PortAudio. I was trying to build a video player in Python. This was a dependency of the library I was using. I gave up on that idea when I found out the app was consuming too much resources. I do not have time to test this out, but if you think the issue has been resolved, you can close it.

**Comment by RossBencina** [comment]
@amoh-godwin no worries, thanks for letting me know. I'll leave this ticket open until someone can verify whether my proposed fix works.
